gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -275,12 +275,27 @@ private WriteObjectResponse doResumableUpload() throws IOException {
 
       boolean objectFinalized = false;
       while (!objectFinalized) {
+
+        if (responseObserver.isComplete()) {
+          // reset streams
+          responseObserver = new InsertChunkResponseObserver(uploadId, writeOffset);
+          call =
+              storageStub
+                  .getChannel()
+                  .newCall(StorageGrpc.getWriteObjectMethod(), stub.getCallOptions());
+          writeObjectRequestStreamObserver =
+              ClientCalls.asyncClientStreamingCall(call, responseObserver);
+          requestStreamObserver =
+              watchdog.watch(
+                  call,
+                  writeObjectRequestStreamObserver,
+                  Duration.ofSeconds(channelOptions.getGrpcWriteMessageTimeoutMillis()));
+        }
+
         WriteObjectRequest insertRequest = null;
         if (requestChunkMap.size() > 0 && requestChunkMap.lastKey() >= writeOffset) {
           insertRequest = getCachedRequest(requestChunkMap, writeOffset);
           writeOffset += insertRequest.getChecksummedData().getContent().size();
-        } else if (requestChunkMap.size() >= channelOptions.getNumberOfBufferedRequests()) {
-          freeUpCommittedRequests(requestChunkMap, writeOffset);
```

Comment:
Will removing this affect the performance?

Reply:
Removing this specifically will not affect the performance. Even it's not functional right now. But yes we do have correctness test plan scheduled for the new logic we are introducing.